### PR TITLE
fix: no cluster admin

### PR DESCRIPTION
**Pull request by f0rmiga**
Improving the RBAC so that Minibroker doesn't operate as cluster-admin. Also, the new API for RBAC allows disabling it completely or making it namespaced so that Minibroker will only be allowed to manage whitelisted namespaces.

### CODE REVIEW

File: charts/minibroker/templates/rbac.yaml [hunk 1]
```diff
@@ -20,8 +57,147 @@ metadata:
...
 - kind: ServiceAccount
+  name: {{ $.Values.rbac.serviceAccount.name | quote }}
+  namespace: {{ $.Release.Namespace | quote }}
+{{- end }}{{/* range $namespace := .Values.rbac.namespaced.whitelist */}}
+{{- else }}{{/* if .Values.rbac.namespaced.enabled */}}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: minibroker
```
Comment (mook-as) [inline charts/minibroker/templates/rbac.yaml:138]: Do we need to worry about having multiple minibrokers deployed into one cluster (possibly across different kubecf / svcat instances)? (i.e. should `.Release.Name` be involved in some way?)

File: charts/minibroker/templates/rbac.yaml [hunk 2]
```diff
@@ -1,12 +1,49 @@
...
+metadata:
+  name: {{ .Values.rbac.serviceAccount.name | quote }}
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+    {{- if gt (len .Values.rbac.serviceAccount.labels) 0 }}
+    {{- .Values.rbac.serviceAccount.labels | toYaml | nindent 4 }}
+    {{- end }}{{/* if gt (len .Values.rbac.serviceAccount.labels) 0 */}}
```
[the remark this answers is not in the document]
Comment (f0rmiga) [reply charts/minibroker/templates/rbac.yaml:15]: I like the first option, but it needs to get the correct syntax.
```
    {{- range $k, $v := .Values.rbac.serviceAccount.labels }}
    {{- printf "%q:%q" $k $v | nindent 4 }}
    {{- end }}
```

File: charts/minibroker/templates/rbac.yaml [hunk 3]
```diff
@@ -20,8 +57,147 @@ metadata:
...
+  name: minibroker
+subjects:
+- kind: ServiceAccount
+  name: {{ .Values.rbac.serviceAccount.name | quote }}
+  namespace: {{ .Release.Namespace | quote }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: minibroker
```
[the remark this answers is not in the document]
Comment (f0rmiga) [reply charts/minibroker/templates/rbac.yaml:69]: Yes. This is needed to access the configmaps in the Minibroker namespace itself.

File: charts/minibroker/templates/rbac.yaml (hunk 1 above)
Comment (f0rmiga) [reply charts/minibroker/templates/rbac.yaml:138]: Good catch. We do need to get `.Release.Name` into the cluster role name.

File: ci/test_integration.sh [hunk 4]
```diff
@@ -17,5 +17,6 @@
 set -o errexit -o nounset -o pipefail
 
 VM_DRIVER=none sudo timeout 3m make create-cluster
+timeout 1m kubectl create namespace minibroker-tests
```
[the remark this answers is not in the document]
Comment (f0rmiga) [reply ci/test_integration.sh:20]: The CI VMs are ephemeral. We don't really care about cleaning up.

File: charts/minibroker/templates/rbac.yaml (hunk 1 above)
Comment (f0rmiga) [reply charts/minibroker/templates/rbac.yaml]: Fixed with https://github.com/kubernetes-sigs/minibroker/pull/65/commits/6f7675e2984304cd239bbe73018430cf371a5237.

File: charts/minibroker/templates/rbac.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (f0rmiga) [reply charts/minibroker/templates/rbac.yaml]: Fixed with https://github.com/kubernetes-sigs/minibroker/pull/65/commits/6f7675e2984304cd239bbe73018430cf371a5237.

**Pull request merged**